test6.go

```diff
@@ -1,61 +1,53 @@
 package main
 
 import (
-    "fmt"
-    "io/ioutil"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
 	"net/http"
 	"os"
-	"log"
-	"io"
-	"encoding/json"
-	"encoding/csv"
 )
 
-type Sheet struct{
-	Sitename string `json:"sitename"`
-	Location string `json:"location"`
-	Address *Address `json:address`
-
-}
-
-type Address struct {
-	Number string `json:"number"`
+func closeFile(f *os.File) {
+	err := f.Close()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %v\n", err)
+		os.Exit(1)
+	}
 }
-
-
-
 func main() {
 
-	csvFile, _ := os.Open("test.csv")
-	reader:= csv.NewReader(csvFile)
-	var sheet [] Sheet
+	m := make(map[string]int)
+
+	csvFile, err := os.Open("test.csv")
+	defer closeFile(csvFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	reader := csv.NewReader(csvFile)
 	for {
-	 line, error:= reader.Read()
-		if error==io.EOF {
+		line, err := reader.Read()
+		if err == io.EOF {
 			break
-		} else if error!= nil{
-			log.Fatal(error)
+		} else if err != nil {
+			log.Fatal(err)
 		}
-		response, err:= http.Get("http://ip-api.com/json/www.google.com?fields=country")
-		/*I  want to able to pass line[0] which is the website in my test csv in the url*
-		Something like http.Get("http://ip-api.com/json/{line[0}?fields=country") only returns empty JSON {}*/
-		if err!= nil{
-		fmt.Println("HTTP request failed with error")
-		} else{	
-			data, _:= ioutil.ReadAll(response.Body)
+
+		response, err := http.Get(fmt.Sprintf("http://ip-api.com/json/%s?fields=country", line[0]))
+		defer response.Body.Close()
```

Comment:
Never defer within a loop, defer is not called at the end of a loop, but only at the end of functions.
Simply write a function that gets the data from the URL and returns a map or something.